src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java

```diff
@@ -103,7 +105,68 @@ public void mainRouteBuilder_routeDirectAddUsernameHeader_BadContentGiven_Should
         });
 
         //Then
-        assertThat(result.getIn().getHeader("analysisUsername")).isNull();
+        assertThat(result.getIn().getHeader(RouteBuilderExceptionHandler.USERNAME)).isNull();
+        camelContext.stop();
+    }
+
+    @Test
+    public void mainRouteBuilder_xRhIdentityHeaderProcessor_givenInvalidHeader_shouldSetNullIdentityHeader() throws Exception {
+        // Given
+        camelContext.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                from("direct:test-xRhIdentityProcessor")
+                        .routeId("test-xRhIdentityProcessor")
+                        .process(MainRouteBuilder.xRhIdentityHeaderProcessor);
+            }
+        });
+
+        //When
+        camelContext.start();
+        camelContext.startRoute("test-xRhIdentityProcessor");
+
+        String x_rh_identity= Base64.getEncoder().encodeToString("BadContentGiven".getBytes());
+        Map<String, Object> headers = new HashMap<>();
+        headers.put(MainRouteBuilder.X_RH_IDENTITY, x_rh_identity);
+
+        Exchange result = camelContext.createProducerTemplate().request("direct:test-xRhIdentityProcessor",  exchange -> {
+            exchange.getIn().setBody(null);
+            exchange.getIn().setHeaders(headers);
+        });
+
+        //Then
+        assertThat(result.getIn().getHeader(RouteBuilderExceptionHandler.USERNAME)).isNull();
+        camelContext.stop();
+    }
+
+    @Test
+    public void mainRouteBuilder_xRhIdentityHeaderProcessor_givenValidHeader_shouldSetUserIdentityHeaders() throws Exception {
+        // Given
+        camelContext.addRoutes(new RouteBuilder() {
```

Comment:
if I dont understand wronly , you are testing a method **MainRouteBuilder.xRhIdentityHeaderProcessor** . So I dont see any point on creating a Camel context with only 1 route that executes that method.

Either you create a unit test to that method, or you test the real routes (starting only those affecting the new code).

Reply:
+1. I've removed the route that was created inside the test and use camelContext.startRoute("add-username-header"); instead